webhooks/cluster_test.go

```diff
@@ -195,317 +162,166 @@ func TestClusterTopologyValidation(t *testing.T) {
+				WithTopology(builder.ClusterTopology().
+					WithClass("foo").
+					WithVersion("v1.2.3+xyz.2").
+					Build()).
+				Build(),
+			in: builder.Cluster("fooboo", "cluster1").
+				WithTopology(builder.ClusterTopology().
+					WithClass("foo").
+					WithVersion("v1.2.3+xyz.1").
+					Build()).
+				Build(),
 		},
 		{
 			name:      "should return error when duplicated MachineDeployments names exists in a Topology",
 			expectErr: true,
-			in: &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "fooboo",
-				},
-				Spec: clusterv1.ClusterSpec{
-					Topology: &clusterv1.Topology{
-						Class:   "foo",
-						Version: "v1.19.1",
-						Workers: &clusterv1.WorkersTopology{
-							MachineDeployments: []clusterv1.MachineDeploymentTopology{
-								{
-									Name: "aa",
-								},
-								{
-									Name: "aa",
-								},
-							},
-						},
-					},
-				},
-			},
+			in: builder.Cluster("fooboo", "cluster1").
+				WithTopology(builder.ClusterTopology().
+					WithClass("foo").
+					WithVersion("v1.19.1").
+					WithMachineDeployment(
+						builder.MachineDeploymentTopology("aa").Build()).
+					WithMachineDeployment(
+						builder.MachineDeploymentTopology("aa").Build()).
+					Build()).
+				Build(),
 		},
 		{
 			name:      "should pass when MachineDeployments names in a Topology are unique",
 			expectErr: false,
-			in: &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "fooboo",
-				},
-				Spec: clusterv1.ClusterSpec{
-					Topology: &clusterv1.Topology{
-						Class:   "foo",
-						Version: "v1.19.1",
-						Workers: &clusterv1.WorkersTopology{
-							MachineDeployments: []clusterv1.MachineDeploymentTopology{
-								{
-									Name: "aa",
-								},
-								{
-									Name: "bb",
-								},
-							},
-						},
-					},
-				},
-			},
+			in: builder.Cluster("fooboo", "cluster1").
+				WithTopology(builder.ClusterTopology().
+					WithClass("foo").
+					WithVersion("v1.19.1").
+					WithMachineDeployment(
+						builder.MachineDeploymentTopology("aa").Build()).
+					WithMachineDeployment(
+						builder.MachineDeploymentTopology("bb").Build()).
+					Build()).
+				Build(),
 		},
 		{
 			name:      "should return error on update when Topology class is changed",
 			expectErr: true,
-			old: &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "fooboo",
-				},
-				Spec: clusterv1.ClusterSpec{
-					InfrastructureRef: &corev1.ObjectReference{},
-					Topology: &clusterv1.Topology{
-						Class:   "foo",
-						Version: "v1.19.1",
-					},
-				},
-			},
-			in: &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "fooboo",
-				},
-				Spec: clusterv1.ClusterSpec{
-					InfrastructureRef: &corev1.ObjectReference{},
-					Topology: &clusterv1.Topology{
-						Class:   "bar",
-						Version: "v1.19.1",
-					},
-				},
-			},
-		},
-		{
-			name:      "should return error on update when Topology version is downgraded",
```

Comment:
Is there a reason why we dropping this test?

Reply:
I think this test is made fully  redundant given the tests on https://github.com/kubernetes-sigs/cluster-api/pull/5622/files#diff-3433b5a0b5e9feccee1de41b6e0c8446d02522f7acdd45ae15203869681ddc2aR212-R224

Let me know if I've done that in error.

Reply:
@killianmuldoon You are right. We can drop this test. It is a duplicate of the other test you linked.